src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -468,6 +468,17 @@ public static URI getResourceFromClassPath(String filename) throws CheckstyleExc
         return uri;
     }
 
+    /**
+     * Finds a resource with a given name in the Checkstyle resource bundle.
+     * This method is intended only for internal use in Checkstyle and tests.
+     *
+     * @param name name of the desired resource
+     * @return URI of the resource
+     */
+    /* package */ static URL getCheckstyleResource(String name) {
```

Comment:
what blocks us to make it public ?

it is util method, context less function.

I still do not like "package" visibility

Reply:
Nothing blocks us. But making this method public, we provide API that is not claimed by anyone.
Andrei [suggested](https://github.com/checkstyle/checkstyle/pull/10947#discussion_r758722745) to make it package private. I agree with them. There is no method where this code could be reused in the Checkstyle sources.
If you insist, I can make this method public.

Reply:
>  But making this method public, we provide API that is not claimed by anyone.

it is not an API, our API is only "api" folder. All other classes are our implementations and we will change them at any time, without warnings.
Lets make it public.

please also update `This method is intended only for internal use in Checkstyle and tests.`
to 
`This method is intended only for internal use in Checkstyle tests for easy mocking to gain 100% coverage.`

we need to be clear on method reason.

Reply:
done